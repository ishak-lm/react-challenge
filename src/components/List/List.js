import React from 'react'
import PropTypes from 'prop-types'

import S from './List.styles'

const List = ({ title, pair, children }) => (
  <S.Container>
    <S.Header>
      <S.Title>{title}</S.Title>
      <S.Pair>{pair}</S.Pair>
    </S.Header>
    <S.Ul>{children}</S.Ul>
  </S.Container>
)

List.propTypes = {
  title: PropTypes.string.isRequired,
  pair: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

export default List
