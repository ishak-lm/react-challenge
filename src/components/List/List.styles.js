import styled from 'styled-components'

import { colors } from '../../constants'

const Container = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;
  align-items: center;
`

const Header = styled.div`
  border-bottom: grey 1px solid;
  width: 95%;
`

const Title = styled.span``

const Pair = styled.span``

const Ul = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0;
  width: 380px;
  height: 800px;
  background-color: rgb(${colors.background.list});
`

export default {
  Ul,
  Container,
  Header,
  Title,
  Pair,
}
