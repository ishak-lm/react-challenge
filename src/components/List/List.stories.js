import React from 'react'
import { withKnobs, text } from '@storybook/addon-knobs'

import List from './List'

export default {
  title: '02 - List',
}

export const StoryList = () => (
  <List title={text('Title', 'Book')} pair={text('Pair', 'ETH/BTC')}>
    <li>Storybook</li>
  </List>
)

StoryList.story = {
  decorators: [withKnobs],
  parameters: { options: { selectedPanel: 'storybook/knobs/panel' } },
}
