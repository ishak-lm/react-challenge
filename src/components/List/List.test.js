/* global describe it expect */
import React from 'react'
import { mount } from 'enzyme'

import List from './List'

describe('List', () => {
  const wrapper = mount(
    <List title="Story" pair="ETH/BTC">
      <li>Hello</li>
    </List>,
  )

  expect(wrapper).toMatchSnapshot()

  it('should have the correct default props', () => {
    expect(wrapper.contains(<li>Hello</li>)).toEqual(true)
  })
})
