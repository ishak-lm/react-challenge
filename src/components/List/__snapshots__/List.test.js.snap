// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<List
  pair="ETH/BTC"
  title="Story"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fdyzsU",
            "rules": Array [
              "
  width: 380px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgb(",
              "29, 38, 44",
              ");
  margin-bottom: 25px;

  @media (max-width: 400px) {
    width: 100%;
    max-width: 380px;
    min-width: 200px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa fdyzsU"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "kEpaif",
                  "rules": Array [
                    "
  border-bottom: grey 1px solid;
  width: 95%;
  padding: 10px 0px 10px 0px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH kEpaif"
            >
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "lastClassName": "cszHku",
                        "rules": Array [
                          "
  margin-right: 20px;
  margin-left: 10px;
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-htpNat cszHku"
                  >
                    Story
                  </span>
                </StyledComponent>
              </styled.span>
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "ietrag",
                        "rules": Array [
                          "
  color: ",
                          "grey",
                          ";
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-bxivhb ietrag"
                  >
                    ETH/BTC
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.ul>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "bIOwfp",
                  "rules": Array [
                    "
  list-style-type: none;
  padding: 0;
  margin: 0;
  width: 100%;
  height: 800px;
  overflow: auto;
",
                  ],
                },
                "displayName": "styled.ul",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              className="sc-ifAKCX bIOwfp"
            >
              <li>
                Hello
              </li>
            </ul>
          </StyledComponent>
        </styled.ul>
      </div>
    </StyledComponent>
  </styled.div>
</List>
`;
